survival_lib.py

```
@@ -176,63 +213,147 @@ def fit_km_logrank(df, duration_col, event_col, group_col):
+            elif len(groups) > 2 and group_col:
+                # OPTIMIZATION: Local computation - call multivariate_logrank_test directly
+                res = multivariate_logrank_test(data[duration_col], data[group_col], data[event_col])
+                stats_data = {
+                    'Test': 'Log-Rank (Multivariate)',
+                    'Statistic': res.test_statistic,
+                    'P-value': res.p_value,
+                    'Comparison': 'All groups'
+                }
+            else:
+                stats_data = {'Test': 'None', 'Note': 'Single group or no group selected'}
+        except Exception as e:
```

Ruff (0.14.10)

Do not catch blind exception: `Exception`

(BLE001)